### PR TITLE
Ghosts can now possess borers if said borers have no ckey assigned.

### PR DESCRIPTION
**Pull request by ghost**
This "fixes" #7541 by giving ghosts a chance to grab a borer during an infestation event if they get there before the valid squad, and in the off chance somebody gets pulled into a borer from egg that doesn't want it so they ghost, some other ghost who does want to be a borer can jump in instead.

Tested, shouldn't be any problems (Tried as a player observer, got in. Tried as a player who had just ghosted, couldn't get in)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/borer.dm [hunk 1]
```diff
@@ -829,3 +829,13 @@ var/global/borer_unlock_types = typesof(/datum/unlockable/borer) - /datum/unlock
...
 	else
 		to_chat(src, "<span class='notice'>No significant chemical agents found in [host]'s blood.</span>")
+
+
+/mob/living/simple_animal/borer/attack_ghost(var/mob/dead/observer/O)
+	if(!(src.key))
+		if(O.can_reenter_corpse)
+			var/response = alert(O,"Do you want to take it over?","This borer has no soul","Yes","No")
+			if(response == "Yes")
+				src.key = O.key
```
Comment (PJB3005) [inline code/modules/mob/living/simple_animal/borer.dm]: Doing it dirtily like this is bound to have issues: you'll need to use the borer's mind / make a new mind at the very least, and use correct transfer procs.

Comment (ghost) [inline code/modules/mob/living/simple_animal/borer.dm]: @pjb3005 How would you suggest this be done? Took me 2 days just to figure out I had one tab too many and that was causing everything to go funky

Comment (clusterfack) [inline code/modules/mob/living/simple_animal/borer.dm]: Find whatever code is handled elsewhere to make a borer and use that here

**Comment by ghost** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

**Comment by ghost** [comment]
Hang on, gotta changelog

**Comment by Kurfursten** [comment]
Will you be able to enter a borer if your body is gibbed?

File: code/modules/mob/living/simple_animal/borer.dm [hunk 2]
```diff
@@ -829,3 +829,13 @@ var/global/borer_unlock_types = typesof(/datum/unlockable/borer) - /datum/unlock
 			to_chat(src, "<span class='notice'>No active chemical agents found in [host]'s blood.</span>")
 	else
 		to_chat(src, "<span class='notice'>No significant chemical agents found in [host]'s blood.</span>")
+
+
+/mob/living/simple_animal/borer/attack_ghost(var/mob/dead/observer/O)
+	if(!(src.key))
+		if(O.can_reenter_corpse)
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/borer.dm]: What if you've been gibbed?

File: code/modules/mob/living/simple_animal/borer.dm [hunk 3]
```diff
@@ -829,3 +829,13 @@ var/global/borer_unlock_types = typesof(/datum/unlockable/borer) - /datum/unlock
...
+			var/response = alert(O,"Do you want to take it over?","This borer has no soul","Yes","No")
+			if(response == "Yes")
+				src.transfer_personality(O.client)
```
Comment (d3athrow) [inline code/modules/mob/living/simple_animal/borer.dm]: Needs sanity check for existing key here, if multiple people are attempting to possess the borer.

Comment (ghost) [inline code/modules/mob/living/simple_animal/borer.dm]: What exactly do you mean? If multiple people are attempting to possess the borer at the same time? If one gets in while another's still got the alert up?

Comment (PJB3005) [inline code/modules/mob/living/simple_animal/borer.dm]: @Skullyton it's possible that whilst somebody's alert is still unanswered, somebody else already accepted.

Comment (ghost) [inline code/modules/mob/living/simple_animal/borer.dm]: Alrighty. Coming up

File: code/modules/mob/living/simple_animal/borer.dm [hunk 4]
```diff
@@ -829,3 +829,13 @@ var/global/borer_unlock_types = typesof(/datum/unlockable/borer) - /datum/unlock
 			to_chat(src, "<span class='notice'>No active chemical agents found in [host]'s blood.</span>")
 	else
 		to_chat(src, "<span class='notice'>No significant chemical agents found in [host]'s blood.</span>")
+
+
+/mob/living/simple_animal/borer/attack_ghost(var/mob/dead/observer/O)
+	if(!(src.key))
+		if(O.can_reenter_corpse)
+			var/response = alert(O,"Do you want to take it over?","This borer has no soul","Yes","No")
+			if(response == "Yes")
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/borer.dm]: So basically 

```
if(response == "Yes" && !src.key)
```

**Pull request merged**